pubnub/lib/src/core/logging/logging.dart

```diff
@@ -130,3 +175,189 @@ abstract class Level {
+
+        // Log keyset information
+        messageString += '\n\tKeysets:';
+        var allKeysets = keysets.keysets;
+        if (allKeysets.isEmpty) {
+          messageString += '\n\t  No keysets configured';
+        } else {
+          // Get all keyset names by trying to access them
+          var keysetNames = <String>[];
+          try {
+            // Try to get the default keyset name by checking which one is the default
+            var defaultKeyset = keysets.defaultKeyset;
+            for (var keyset in allKeysets) {
+              // Find the name by comparing with default
+              if (keyset == defaultKeyset) {
+                keysetNames.add('default');
+              } else {
+                // For non-default keysets, we'll use a generic name
+                keysetNames.add('keyset_${keysetNames.length + 1}');
+              }
+            }
+          } catch (e) {
+            // If no default keyset, just number them
+            for (var i = 0; i < allKeysets.length; i++) {
+              keysetNames.add('keyset_${i + 1}');
+            }
+          }
+
+          for (var i = 0; i < allKeysets.length; i++) {
+            var keyset = allKeysets[i];
+            var name = keysetNames[i];
+            messageString += '\n\t  $name:';
+            messageString += '\n\t    Subscribe Key: ${keyset.subscribeKey}';
+            messageString +=
+                '\n\t    Publish Key: ${keyset.publishKey ?? 'not provided'}';
+            messageString +=
+                '\n\t    Secret Key: ${keyset.secretKey != null ? 'provided' : 'not provided'}';
+            messageString += '\n\t    User ID: ${keyset.userId}';
+            messageString +=
+                '\n\t    Auth Key: ${keyset.authKey != null ? 'provided' : 'not provided'}';
+            messageString +=
+                '\n\t    Cipher Key: ${keyset.cipherKey != null ? 'provided' : 'not provided'}';
+            if (keyset.settings.isNotEmpty) {
+              messageString += '\n\t    Settings: ${keyset.settings}';
+            }
+            if (i == 0 && keysetNames[i] == 'default') {
+              messageString += '\n\t    (Default)';
+            }
+          }
+        }
+      } else if (detailsType == LogEventDetailsType.apiParametersInfo) {
+        // Handle both Map and Object types
+        Map<String, dynamic> detailsMap;
+        if (details is Map) {
+          detailsMap = details as Map<String, dynamic>;
+        } else if (details != null) {
+          // Extract properties from any object
+          detailsMap = extractObjectProperties(details!);
+        } else {
+          detailsMap = {};
+        }
+        messageString +=
+            '\n\t${detailsMap.entries.map((entry) => '${entry.key}: ${entry.value}').join('\n\t')}';
+      } else if (detailsType == LogEventDetailsType.networkRequestInfo) {
+        var requestMap = details as Map<String, dynamic>;
+
+        for (var entry in requestMap.entries) {
```

Comment:
It works,  although I do think this may be challenging to maintain in the long-run. Do you have the ability to introduce and log more abstract types, let's say, `NetworkRequest` or `NetworkResponse`? They can have properties specific to request and response, respectively, and override `toString()` method. Then, you could always call `.toString()` regardless of the passed object.